interface/backend/cases/models.py

```diff
@@ -18,6 +60,104 @@ class Case(models.Model):
 
     series = models.ForeignKey('images.ImageSeries', related_name='cases')
 
+    kVp = models.IntegerField(default=None)
+    mA = models.IntegerField(default=None)
+    DLP = models.IntegerField(default=None)
+
+    SCREENING_VISIT_CHOICES = (
+        ("BL", "Baseline"),
+        ("Y1", "Year 1"),
+        ("Y2", "Year 2")
+    )
+
+    screening_visit = models.CharField(max_length=2, choices=SCREENING_VISIT_CHOICES, default="BL")
+    clinical_information = models.CharField(max_length=250, default="Lung cancer screening.")
+    comparison = models.CharField(max_length=250, default="None.")
+
+    DIAGNOSTIC_QUALITY_CHOICES = (
+        ("SF", "Satisfactory"),
+        ("LI", "Limited, but interpretable"),
+        ("ND", "Non - diagnostic")
+    )
+
+    diagnostic_quality = models.CharField(max_length=2, choices=DIAGNOSTIC_QUALITY_CHOICES, default="SF")
+    exam_parameters_comment = models.CharField(max_length=250, default="")
+
+    @unique  # ensures all variables are unique
+    @django_enum
+    class ShapeChoices(IntEnum):
+        NONE = 0
+        MILD = 1
+        MODERATE = 2
+        SEVERE = 3
+
+    @unique  # ensures all variables are unique
+    @django_enum
+    class HeartShapeChoices(IntEnum):
+        NORMAL = 0
+        MILDLY_ENLARGED = 1
+        MODERATELY_ENLARGED = 2
+        SEVERELY_ENLARGED = 3
+
+    COPD = models.IntegerField(
+        choices=[(choice.value, choice.name.replace("_", " ")) for choice in ShapeChoices],
+        default=ShapeChoices.NONE.value)
+
+    fibrosis = models.IntegerField(
+        choices=[(choice.value, choice.name.replace("_", " ")) for choice in ShapeChoices],
+        default=ShapeChoices.NONE.value)
+
+    lymph_nodes = models.CharField(max_length=250, default="None.")
+
+    other_findings = models.CharField(max_length=250, default="None.")
+
+    left_pleural_space = models.ForeignKey(PleuralSpace,
+                                           on_delete=models.CASCADE,
+                                           related_name="left_pleural_space",
+                                           null=True)
+    right_pleural_space = models.ForeignKey(PleuralSpace,
+                                            on_delete=models.CASCADE,
+                                            related_name="right_pleural_space",
+                                            null=True)
+
+    heart_size = models.IntegerField(
+        choices=[(choice.value, choice.name.replace("_", " ")) for choice in HeartShapeChoices],
+        default=HeartShapeChoices.NORMAL.value)
+
+    coronary_calcification = models.IntegerField(
+        choices=[(choice.value, choice.name.replace("_", " ")) for choice in ShapeChoices],
+        default=ShapeChoices.NONE.value)
```

Comment:
I worry that we have too many nullable (or quasi-nullable with ``NONE`` enum value) here. These are pretty common sources of bugs. Can I suggest trying a "hanging table" approach and seeing how that feels? eg.

```
class Parent(models.Model):
    field_a = models.IntegerField()
    field_b = models.IntegerField()

class ExtraDataForParent(models.Model):
    parent = models.OneToOneField(Parent, related_name='extra_data', primary_key=True)
    field_c = models.IntegerField()
    field_d = models.IntegerField()
```
Note that crucially  ``field_c`` and ``field_d`` are not nullable as you only create ``ExtraData`` models when you know you have both of those fields, so there is no possibility of having ``field_c`` but not ``field_d`` etc. etfc.

Reply:
Thank you, it's absolutely rational to decompose the models.